doc/s3ip_sysfs/s3ip_sysfs_hld.md

```diff
@@ -0,0 +1,309 @@
+
+### Definitions/Abbreviations 
+| Abbreviation             | Description                                       |
+|--------------------------|---------------------------------------------------|
+| S3IP                     | Simplified Switch System Integration Program      |
+| CPLD                     | Complex Programmable Logic Device                 |
+| I2C                      | Inter-Integrated Circuit                          |
+| SAI                      | Switch Abstraction Interface                      |
+| ODM                      | Original Design Manufacturer
+
+### 1. Overview 
+
+SONiC is designed to be portable to a variety of network devices. Many devices share the same ASIC platform and only differ in device-specific hardware components, such as PSUs, fan modules, and environment sensors. Currently, ODM vendor provides drivers to expose the device-specific hardware through sysfs, allowing SONiC to communicate with them, as described in the [Porting Guide](https://github.com/sonic-net/SONiC/wiki/Porting-Guide). However, many inefficient porting works are still required for SONiC developers due to different drivers from different devices. 
+
+The S3IP sysfs specification defines a unified interface to access peripheral hardware on devices from different vendors, making it easier for SONiC to support different devices and platforms.
+
+**Requirements for S3IP sysfs:**
+1. Should be able to work with the existing platform APIs
+2. The hierarchy of sysfs path structure should be defined in a specification.
+3. The sysfs path in the specification should be defined clearly, thus, platform API and test cases can be reused.
+4. The sysfs directory structure can be generated automatically through the framework.
+5. Support reuse of the old driver sysfs paths without redeveloping new drivers.
+
+S3IP sysfs contains two sections:
+
+- sysfs specification: specifies the sysfs directory structure and file content
+- sysfs framework: generate the sysfs directory structure that conforming to the specification.
+
+### 2. S3IP sysfs specification
+#### 2.1 S3IP sysfs specification scenario
+
+Figure2-1 where S3IP sysfs is used.
+
+![scenario](/images/s3ip_sysfs/s3ip_sysfs_scenario.svg)
+
+The S3IP sysfs specification is represented as an organized sysfs directory structure on white-box devices. Device management software and debugging tools need to access hardware through this interface.
+
+Both vendors and users should comply with the S3IP sysfs specification. Device vendors focus on the specification implementation, while users verify the usability of the device against this specification.
+
+Figure2-2 S3IP sysfs diretory demo
+
+![demo](/images/s3ip_sysfs/s3ip_sysfs_demo.png)
+
+#### 2.2 Design of the S3ip sysfs specification
+Design Principles
+ - Unified, specificationized behavior, consistent experience among all SONiC devices, regardless of underlying platform; 

```

Comment:
I feel the 2.0 platform APIs does the same thing, isn't it? These APIs are platform independent and platform vendors can implement them as they need i.e. 

1. Either read them directly from the SysFS created by device drivers, or 
2. Read them raw and then apply some filters/masks/pre-processing steps to get the final value

What is the reason/motivation behind the SysFS?
With S3IP, I see you are able to manage 1 but what about point#2?